### PR TITLE
[WIP] feat: add e2e tests

### LINKED ISSUE #49: Feature: e2e tests
**Is your feature request related to a problem? Please describe.**
Currently, this module is not tested in an integration, e2e scenario. 

**Describe the solution you'd like**
We need to implement e2e testing. 

- Create top-level folder named `e2e`
- Generate nest application using nest cli inside the `e2e` folder named `test-app`
- Define an import alias for `@nestjs-addons/in-memory-db` in the apps `tsconfig.json` file, pointing the path up to the parent `lib/` directory. So that we can simulate real-world use.
- Update `app.module.ts` to import the `InMemoryDBModule` and register as `forRoot`
- Update `app.controller.ts` to do the following:
  - Inject `InMemoryDBService` into the constructor
  - Call the `createMany` method in the constructor to generate several example records
  - Define actions/endpoints for *all* public methods on the service, both sync and async
- Update the included nested `e2e` folder with tests to verify that each action/endpoint returns the expected values.

### PR DESCRIPTION
**Pull request by jordanpowell88**
Closes #49

### CODE REVIEW

**Comment by jordanpowell88** [comment]
@wesleygrimes did you want me to try to get these tests to run from root or did we want to look into adding nx?

**Comment by jordanpowell88** [comment]
I am able to run tests in the root of the project locally just fine. Not sure why it says it can't find supertest even though it is a devDependency. Screen Shot 2019-08-28 at 7 21 20 PM

**Comment by wesleygrimes** [comment]
Let’s pair on this tomorrow

**Comment by jordanpowell88** [comment]
Sounds good

**Review by wesleygrimes — CHANGES_REQUESTED** [review]
Need to add coverage folder to nested .gitignore

**Comment by wesleygrimes** [comment]
Looks like the files were added to the .gitignore, but not actually removed from the commit. Can you rebase on master?

**Comment by jordanpowell88** [comment]
I have another branch that has e2e tests for using forFeature and forRoot as well

File: e2e/test-app/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<p align="center">
```
Comment (itayod) [inline e2e/test-app/README.md:1]: Do we need this README file?

File: e2e/test-app/coverage/clover.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (itayod) [inline e2e/test-app/coverage/clover.xml:1]: please remove all files under `e2e/test-app/coverage` :)

**Review by itayod — COMMENTED** [review]
@jordanpowell88 Nicely done!! great work 👍

File: e2e/test-app/README.md (hunk 1 above)
Comment (jordanpowell88) [reply e2e/test-app/README.md:1]: Probably not, I can remove it

**Pull request closed without merging**